contracts/SpokePool.sol

```diff
@@ -862,6 +908,82 @@ abstract contract SpokePool is
         );
     }
 
+    /******************************************
+     *         USS RELAYER FUNCTIONS          *
+     ******************************************/
+
+    function fillRelayUSS(
+        address depositor,
+        address recipient,
+        address exclusiveRelayer,
+        address inputToken,
+        address outputToken,
+        uint256 inputAmount,
+        uint256 outputAmount,
+        uint256 repaymentChainId,
+        uint256 originChainId,
+        uint32 depositId,
+        uint32 fillDeadline,
+        bytes memory message
+    ) public nonReentrant unpausedFills {
+        USSLib.RelayExecution memory relayExecution = USSLib.RelayExecution({
+            relay: USSLib.RelayData({
+                depositor: depositor,
+                recipient: recipient,
+                relayer: exclusiveRelayer,
+                inputToken: inputToken,
+                outputToken: outputToken,
+                inputAmount: inputAmount,
+                outputAmount: outputAmount,
+                originChainId: originChainId,
+                destinationChainId: chainId(),
+                depositId: depositId,
+                expiryTimestamp: fillDeadline,
+                message: message
+            }),
+            relayHash: bytes32(0),
+            updatedOutputAmount: outputAmount,
+            updatedRecipient: recipient,
+            updatedMessage: message,
+            repaymentChainId: repaymentChainId,
+            executionType: USSLib.RelayExecutionType.FastFill,
+            payoutAdjustmentPct: 0
+        });
+        relayExecution.relayHash = keccak256(abi.encode(relayExecution.relay));
+
+        // Validate
```

Comment:
Does this include evaluating `exclusiveRelayer` against `msg.sender` ? If so, would that instead be better placed at the very top of the function?

Reply:
yep 912db111aa03ccebd0a953f058cba0a8aff0985c